components/form/Select.tsx

```diff
@@ -0,0 +1,247 @@
+import React from "react";
+import { useFormContext } from "react-hook-form";
+import classNames from "classnames";
+import ToolTip from "./Tooltip";
+import { FEEDBACK } from "@/utils/constant";
+import { ErrorMessage } from "./ErrorMessage";
+import { Label } from "./Label";
+import { showError, showSuccess } from "../../utils/form-helpers";
+import { ArrowDown2, InfoCircle, TickCircle } from "iconsax-react";
+import { CommonTypes, SelectFieldProps, TooltipTypes } from "@/types/index";
+
+export function SelectField({
+  children,
+  className,
+  containerClassName,
+  floatLabel,
+  feedBack,
+  label,
+  labelClassName,
+  leadingIcon,
+  name,
+  toolTip,
+  hintText,
+  toolTipTitle,
+  toolTipContent,
+  toolTipColor,
+  toolTipPlacement,
+  toolTipIcon,
+  showTooltipArrow,
+}: SelectFieldProps) {
+  const {
+    register,
+    formState: { dirtyFields, errors },
+  } = useFormContext();
+
+  const hasErrors = !!errors?.[name];
+  const isValid = !!dirtyFields?.[name] && !hasErrors;
+
+  return (
+    <>
+      {label && !floatLabel && (
+        <Label
+          feedBack={FEEDBACK.NONE}
+          className="text-base flex items-center gap-x-2"
+          name={name}
+          htmlFor={name}
+          text={label}
+        >
+          {toolTip && (
+            <ToolTip
+              arrow={showTooltipArrow}
+              title={toolTipTitle}
+              content={toolTipContent}
+              placement={toolTipPlacement}
+              color={toolTipColor}
+              toolTipIcon={toolTipIcon}
+            />
+          )}
+        </Label>
+      )}
+      <div
+        className={classNames(
+          "relative z-0 mb-2 w-full group",
+          containerClassName
+        )}
+      >
+        {leadingIcon && (
+          <div className="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none">
+            {leadingIcon}
+          </div>
+        )}
+        <select
+          id={name}
+          {...register(name)}
+          className={classNames(
+            "block py-3.5 px-2 w-full text-base appearance-none transition ease-in-out bg-secondary-25 border-secondary-100 border-2 rounded-md focus:outline-none focus:ring-0 peer",
+            className,
+            { "pl-10 pr-8": leadingIcon },
+            {
+              "mt-6 placeholder-transparent border-secondary-300 appearance-none bg-secondary-25 text-secondary-700 focus:ring-0 peer":
+                floatLabel,
+            },
+            {
+              errorClassName: showError(hasErrors, feedBack),
```

Comment:
I am currently not sure of your implementation here. Do we want to write `errorClassName` when there is an error or write some error class name

Reply:
resolve